crates/astria-conductor/src/executor/mod.rs

```diff
@@ -214,6 +231,37 @@ impl<C: ExecutionClient> Executor<C> {
             return Ok(Some(execution_hash.clone()));
         }
 
+        let mut final_response = Ok(Some(self.execution_state.clone()));
+        // TODO: (GHI 250 - https://github.com/astriaorg/astria/issues/250):
+        // if blocks are coming from DA, do we still need them to pass through
+        // the queue? if not, we can skip the insert and pop_blocks calls
+        if let Ok(None) = self.block_queue.insert(block.clone()) {
+            return final_response;
+        };
+        // TODO: (GHI 250 - https://github.com/astriaorg/astria/issues/250):
+        // add a match statement here to either `pop_blocks` (returns soft and
+        // head) or `pop_soft_blocks` (just soft) based on the `execution_commit_level` setting
+        let queued_blocks = self.block_queue.pop_blocks();
+
+        if let Some(blocks) = queued_blocks {
+            // execute all the blocks returned from the queue
+            for block in blocks {
+                final_response = self.execute_single_block(block).await;
+            }
+        }
+
+        final_response
+    }
+
+    /// Send a block to the execution layer for execution.
+    ///
+    /// This function takes a SequencerBlockSubset and its associated data to
+    /// build the inputs for a call to `DoBlock` on the execution layer.
+    async fn execute_single_block(
+        &mut self,
+        block: SequencerBlockSubset,
+    ) -> Result<Option<Vec<u8>>> {
+        // get the previous execution state to pass to do block
         let prev_block_hash = self.execution_state.clone();
```

Comment:
this is going to error if 2 blocks at the same height are executed one after the other; from what i understand `self.block_queue.pop_blocks()` can return multiple blocks at the same height, this should handle that